application/classes/Ushahidi/Listener/PostListener.php

```diff
@@ -0,0 +1,51 @@
+<?php defined('SYSPATH') or die('No direct script access');
+
+/**
+ * Ushahidi PostSet Listener
+ *
+ * Listens for new posts that are added to a set
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use League\Event\AbstractListener;
+use League\Event\EventInterface;
+
+use Ushahidi\Core\Entity\WebhookQueueRepository;
+use Ushahidi\Core\Entity\WebhookRepository;
+
+class Ushahidi_Listener_PostListener extends AbstractListener
+{
+	protected $repo;
+
+	protected $webhook_repo;
+
+	public function setRepo(WebhookQueueRepository $repo)
```

Comment:
These could probably just be done through constructor injection.. matter of opinion though